packages/SwingSet/src/devices/timer/device-timer.js

```diff
@@ -132,24 +149,28 @@ function makeTimerMap(state = undefined) {
   }
 
   // We don't expect this to be called often, so we don't optimize for it.
-  function remove(targetHandler) {
+  /**
+   *
+   * @param {Waker} target
+   * @returns {bigint[]}
+   */
+  function remove(target) {
+    /** @type {bigint[]} */
     const droppedTimes = [];
     let i = 0;
     while (i < schedule.length) {
       const { time, handlers } = schedule[i];
       if (handlers.length === 1) {
-        if (handlers[0].handler === targetHandler) {
+        if (handlers[0].handler === target) {
           schedule.splice(i, 1);
           droppedTimes.push(time);
         } else {
           i += 1;
         }
       } else {
         // Nothing prevents a particular handler from appearing more than once
-        for (const { handler } of handlers) {
-          // @ts-expect-error xxx Waker vs IndexedHandler
-          if (handler === targetHandler && handlers.indexOf(handler) !== -1) {
-            // @ts-expect-error xxx Waker vs IndexedHandler
+        for (const handler of handlers) {
+          if (handler.handler === target && handlers.indexOf(handler) !== -1) {
```

Comment:
I don't know why it's checking for presence of the value it just got out of the iterator

Reply:
Well that's concerning. Previously `handler` was the `handlerRecord.handler` which could never be present in the `handlers` array, so this condition would have never matched. I am wondering if we somehow never encountered a case where more than one handler appeared in the `handlers` list?

Reply:
Ok I see that the new test is tripping this case before the fix. As a matter of commit stacking, I would have preferred to see the commit adding the test marking it as `.failing`, and for the fix to remove the `.failing`.

Reply:
I believe this is still incorrect. splicing an array while iterating over it will move the items such that the item following the one being sliced out will be skipped.

The only safe way is iterating in reverse using the index.